components/collector/src/source_collectors/junit/tests.py

```diff
@@ -39,5 +39,8 @@ def _include_entity(self, entity: Entity) -> bool:
     @staticmethod
     def __entity(case_node: Element, case_result: str) -> Entity:
         """Transform a test case into a test case entity."""
+        class_name = case_node.get("classname", "")
         name = case_node.get("name", "<nameless test case>")
-        return Entity(key=name, name=name, class_name=case_node.get("classname", ""), test_result=case_result)
+        key = f"{class_name}:{name}"
+        old_key = name  # Key was changed after Quality-time 5.16.1, enable migration of user entity data
```

Comment:
How is the data migrated when it's the same value as `name`?

Reply:
The data is migrated by the collector when creating a measurement in https://github.com/ICTU/quality-time/blob/8536d0bd1075f042ad4b47e7e59c6a682beacc85/components/collector/src/database/measurements.py#L16 via `copy_entity_user_data`. Now sure what you mean by 'when it's the same value as `name`'?

Reply:
I don't see where `old_key` is being used, I thought that perhaps you intended to write a migration that wasn't included in this PR (yet)? But if the value of `old_key` is the same as the value of `name`, then I don't see why this migration wouldn't use `name` instead of `old_key` directly

Reply:
The `old_key` is used in the `Source.copy_entity_user_data` method which is called by `Measurement.copy_entity_user_data` which in turn is called by `create_measurement` when a new measurement is created.